hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/SecurityAutoConfiguration.java

```diff
@@ -33,18 +41,42 @@
  * {@link EnableAutoConfiguration Auto-configuration} for security.
  */
 @Configuration
-@EnableConfigurationProperties({ DdiSecurityProperties.class, HawkbitSecurityProperties.class })
+@EnableConfigurationProperties({SecurityProperties.class, DdiSecurityProperties.class, HawkbitSecurityProperties.class})
 public class SecurityAutoConfiguration {
 
     /**
      * @return the {@link TenantAware} singleton bean which holds the current
      *         {@link TenantAware} service and make it accessible in beans which
      *         cannot access the service directly, e.g. JPA entities.
+     *
+     * @param authoritiesResolver
+     *             The user authorities/roles resolver
      */
     @Bean
     @ConditionalOnMissingBean
-    public TenantAware tenantAware() {
-        return new SecurityContextTenantAware();
+    public TenantAware tenantAware(final UserAuthoritiesResolver authoritiesResolver) {
+        return new SecurityContextTenantAware(authoritiesResolver);
+    }
+
+    /**
+     * Creates a {@link UserAuthoritiesResolver} bean that is responsible for resolving user authorities/roles
+     *
+     * @param securityProperties
+     *            The Spring {@link SecurityProperties} for the security user
+     * @param multiUserProperties
+     *            The {@link MultiUserProperties} for the managed users
+     *
+     * @return    an {@link InMemoryUserAuthoritiesResolver} bean
+     */
+    @Bean
+    public UserAuthoritiesResolver inMemoryAuthoritiesResolver(final SecurityProperties securityProperties,
+            final MultiUserProperties multiUserProperties) {
+        final Map<String, List<String>> usersToPermissions = multiUserProperties.getUsers().stream() //
+                .collect(Collectors.toMap(User::getUsername, User::getPermissions));
+
+        usersToPermissions.put(securityProperties.getUser().getName(), securityProperties.getUser().getRoles());
```

Comment:
I guess we need to check if there are any users defined within the multiUserProperties first and add the spring security user only in opposite case. Otherwise if we define the user with the same name as spring security user, he will get spring security roles/permissions (see InMemoryUserManagementAutoConfiguration#userDetailsService)

Reply:
Good point. How about adding the Spring security user only if it is not yet contained in the map AFTER processing multiUserProperties?

Reply:
I think this is still ok, if we define a user with the same username as the spring user, that user will not be taken into consideration. The Spring security user should always be there, and should always have the predefined set of authorities related to him.

What we can do, we can check if a user is defined with the same name as the spring security user, then we can throw an exception for example. But I wouldn't make the spring security user optional

Reply:
As I've already mentioned above, please, take a look at InMemoryUserManagementAutoConfiguration#userDetailsService implementation. It is either-or semantic, meaning that in case we define the users by ourselves via multiUserProperties spring security user will be omitted in user management completely. So in my opinion we should not add the spring security user at all to the map in case multiUserProperties is not empty for the current implementation. Alternatively, we could change the logic in InMemoryUserManagementAutoConfiguration to ALWAYS setup the spring security user

Reply:
ok I was not aware that this is the current behavior of the `InMemoryUserManagement`, somehow was expecting it to configure 'additional' users, and not override the spring security user. I'm fine with either changing the autoconfiguration or adapting the user permissions here

Reply:
ok, I will fix that accordingly to keep the current semantics. Thanks guys!